File: Hotel Management project/hotelSystem/src/app/services/bill.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class BillService {

  constructor(private httpClient:HttpClient) { }

  generateReport(data:any){
    console.log(data);
    return this.httpClient.post("http://localhost:8090/bill/generateReport",data,{headers:new HttpHeaders().set('Content-type',"application/json")});
  }

  getPdf(data:any):Observable<Blob>{
    return this.httpClient.post("http://localhost:8090/bill/getPdf",data,{responseType:'blob'});
  }

  getBills(){
   return  this.httpClient.get("http://localhost:8090/bill/getBills");
  }


  delete(id:any){
    return this.httpClient.post("http://localhost:8090/bill/deleteBill/"+id,{headers:new HttpHeaders().set('Content-Type',"application/json")})
  }
}

File: Hotel Management project/HotelManagementSystemProject/src/main/java/com/in/hotel/Service/BillService.java
package com.in.hotel.Service;

import java.util.List;
import java.util.Map;

import org.springframework.http.ResponseEntity;

import com.in.hotel.model.Bill;

public interface BillService {

	ResponseEntity<String> generateReport(Map<String, Object> requestMap);

	ResponseEntity<List<Bill>> getBills();

	ResponseEntity<byte[]> getPdf(Map<String, Object> requestMap);

	ResponseEntity<String> deleteBill(Integer id);

}
